contents/testing/security-testing.md

```diff
@@ -0,0 +1,93 @@
+- **Risk Assessment**: This testing involves analysis of security risks observed in the organization. Risks are classified as  Low, Medium and High. This testing recommends controls and measures to reduce the risk.
+- **Security Auditing**: This is an internal inspection of Applications and Operating systems for security flaws. An audit can also be done via line by line inspection of code.
+- **Posture Assessment**: This combines Security Scanning, Penetration Testing, and Risk Assessments to show an overall security posture of an organization.
+
+## Why Have Security Testing?
+
+A comprehensive security testing framework deals with validation across all layers of an application. Through this, the organization can evaluate their application code for vulnerabilities and take remedial measures for the same. Recently, many of the software development organizations have been making use of secure software development life cycle methodologies to ensure identification and rectification of vulnerability areas early on in the application development process.
+
+Security Testing helps to ensure:
+- Confidentiality of Information.
+- Integrity of Information.
+- Authenticity of Identity.
+- Availability of information and services.
+- Authorisation: The process of determining whether a requester is allowed to receive a service or perform an operation (e.g. Access Control).
+
+## How to Get Started with Security Testing?
+
+As a subset of software testing, security testing operates in tandem with your SDLC. It is a conscious decision that needs to be made for a more secure application.
+
+![Generic SDLC Model](security-testing/Sec-Test-Fig-1.png "Generic SDLC Model")
+
+<box id="info">:fas-info-circle: Knowledge of Software Development Life-Cycles (SDLCs) would be useful but not necessary. If you would like to find out more about SDLC Processes please click [here](https://nus-cs2103-ay1920s1.github.io/website/se-book-adapted/chapters/processModels.html).</box>
+
+### Phase 1: Before Development Begins
+
+Define a SDLC and review policy standards. Ensure that there are appropriate policies, standards, and documentation in place. Documentation is extremely important as it gives development teams guidelines and policies that they can follow. 
+
+People can only do the right thing if they know what the right thing is.
+
+### Phase 2: During Definition and Design
+
+Review security requirements, always make sure you have a <tooltip content="A security policy is a written document in an organization outlining how to protect the organization from threats, including computer security threats, and how to handle situations when they do occur.">__security policy__</tooltip>! Security requirements define how an application works from a security perspective. It is essential that the security requirements are tested. Testing in this case means testing the assumptions that are made in the requirements and testing to see if there are gaps in the requirements definitions.
```

Comment:
The reader might find it easier to understand this section if there is an example/extract of a security policy to see.